nsenter/event.go

```diff
@@ -767,9 +824,48 @@ func (e *NSenterEvent) processChownSyscallRequest() error {
 	return nil
 }
 
+func (e *NSenterEvent) getProcCreds(pipe *os.File) error {
```

Comment:
Why aren't you using the routines that we have in sysbox-libs for this purpose (i.e. `io.go`) by extending them if needed? Those routines were built making use of the proper abstractions provided by golang's unix pkg, so that we avoid calling syscalls directly. Also, keep in mind that there's already code in sysbox-fs making use of these routines, so we would end up with overlapping logic within the same daemon.

Reply:
Oh I had not noticed; let me see if I can move the code into `io.go` then. You are right that it's better to use `unix` instead of `syscall`.

Reply:
I tried hard to do this, but it's a pain unfortunately. The ipc stuff we have on io.go works using *net.UnixConn, and it turns out that setting SO_PASSCRED on the socket (needed to pass credentials across it) is a pain because there is no straight forward mechanism to do it (we can't use unix.SetSockoptInt() as I was hoping). I'll look into it a bit more, but don't want to spend much more time on this.

Reply:
Rodny and I agreed to do this in a subsequent refactoring.